nemo/collections/tts/models/speechllm/megatron_t5_speechllm_model.py

```diff
@@ -1381,7 +1381,7 @@ def build_virtual_prompt_tarred_dataset(
 
         return dataset, dataloader
 
-    def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> Any:
+    def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0, log_metrics=True, global_step=None) -> Any:
```

Comment:
What is `log_metrics` supposed to do? We have some self.logger.experiments calls that are guarded by this and some that are not. Are all such calls intended to be in an if statement?

Reply:
log_metrics specifies whether we want to log scalars related to TTS evaluation in predict step. The things that are not guarded are audio, images and question texts (also left out some metrics for ASR task that Shubham was doing earlier). Let me know if I should change its name or guard/unguard different metrics.

Reply:
Do you think the images/audio/tests should be or should not be guarded?
Can you add a comment that indicates what it does?